sklearn/mixture/_gaussian_mixture.py

```diff
@@ -314,20 +314,27 @@ def _compute_precision_cholesky(covariances, covariance_type):
         precisions_chol = np.empty((n_components, n_features, n_features))
         for k, covariance in enumerate(covariances):
             try:
-                cov_chol = linalg.cholesky(covariance, lower=True)
+                cov_chol = linalg.cholesky(covariance,
+                                           lower=True,
+                                           check_finite=False)
             except linalg.LinAlgError:
                 raise ValueError(estimate_precision_error_message)
             precisions_chol[k] = linalg.solve_triangular(cov_chol,
                                                          np.eye(n_features),
-                                                         lower=True).T
+                                                         lower=True,
+                                                         check_finite=False).T
```

Comment:
How do we know that `linalg.cholesky` never returns non finite values ?

Reply:
If think that `linalg.cholesky` doesn't return non finite values provided that:
 - `cov_chol` does not contain non-finite values
 - `cov_chol` is positive definite (which is the case assuming the estimation of the matrix is accurate and the data is represented in all the dimensions) 
 - the implementation of the Cholesky factorisation is properly tested on scipy's end.

Again, I need confirmation,  as it might be safer to remove this patch too.

Reply:
`linalg.cholesky` does check if covariance is finite, but I do not know if `cov_chol` is 100% finite. scipy uses lapack's [dpotrf](http://www.netlib.org/lapack/explore-html/d1/d7a/group__double_p_ocomputational_ga2f55f604a6003d03b5cd4a0adcfb74d6.html) to compute the factorization. One would need to investigate to see if this routine would return something non-finite.

Given that, I would be okay to have `check_finite=False` here if we have a benchmark to see how much better this performs. This is because I do think there is much a user can do if the return of `linalg.cholesky` is non-finite and `covariance` is finite and checked by `linalg.cholesky`.

Reply:
I've tried to stress `dpotrf` via `scipy.linalg.decomp_cholesky._cholesky` with various inputs to make it return non-finite results: I haven't managed to make it return such values.

I also have run a [simple benchmark](https://gist.github.com/jjerphan/45e36d9859ea324888a091ba1b37ac9b) to know the performances of setting `check_finite` to `False` for `linalg.cholesky`

```
n_features	check_finite	Running time (100 trials)
--------------------------------------------------------------------------------------------------
10		True		0.0002 ± 0.0005s
10		False		0.0001 ± 0.0001s
--------------------------------------------------------------------------------------------------
50		True		0.0002 ± 0.0001s
50		False		0.0002 ± 0.0007s
--------------------------------------------------------------------------------------------------
100		True		0.0004 ± 0.0002s
100		False		0.0003 ± 0.0002s
--------------------------------------------------------------------------------------------------
500		True		0.0098 ± 0.0030s
500		False		0.0090 ± 0.0025s
--------------------------------------------------------------------------------------------------
1000		True		0.0591 ± 0.0067s
1000		False		0.0540 ± 0.0042s
--------------------------------------------------------------------------------------------------
2000		True		0.3745 ± 0.0534s
2000		False		0.3441 ± 0.0188s
```

Maybe we should revert this `check_finite` to `True` for `linalg.cholesky`?

Reply:
I've tried to stress `dpotrf` via `scipy.linalg.decomp_cholesky._cholesky` with various inputs to make it return non-finite results: I haven't managed to make it return such values.

I also have run a [simple benchmark](https://gist.github.com/jjerphan/45e36d9859ea324888a091ba1b37ac9b) to know the performances of setting `check_finite` to `False` for `linalg.solve_triangular`.

```
n_features	check_finite	Running time (100 trials)
--------------------------------------------------------------------------------------------------
10		True		0.0002 ± 0.0005s
10		False		0.0001 ± 0.0001s
--------------------------------------------------------------------------------------------------
50		True		0.0002 ± 0.0001s
50		False		0.0002 ± 0.0007s
--------------------------------------------------------------------------------------------------
100		True		0.0004 ± 0.0002s
100		False		0.0003 ± 0.0002s
--------------------------------------------------------------------------------------------------
500		True		0.0098 ± 0.0030s
500		False		0.0090 ± 0.0025s
--------------------------------------------------------------------------------------------------
1000		True		0.0591 ± 0.0067s
1000		False		0.0540 ± 0.0042s
--------------------------------------------------------------------------------------------------
2000		True		0.3745 ± 0.0534s
2000		False		0.3441 ± 0.0188s
```

Maybe we should revert this `check_finite` to `True` for `linalg.solve_triangular`? This solved, it can then be merged.